components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementAdminService.java

```diff
@@ -81,21 +90,26 @@ public ServiceProvider getApplication(String applicationName)
      * @return Application Basic information array
      * @throws org.wso2.carbon.identity.application.common.IdentityApplicationManagementException
      */
-    public ApplicationBasicInfo[] getAllApplicationBasicInfo()
-            throws IdentityApplicationManagementException {
-        applicationMgtService = ApplicationManagementService.getInstance();
-
-        ApplicationBasicInfo[] applicationBasicInfos = applicationMgtService.getAllApplicationBasicInfo(getTenantDomain(), getUsername());
-        ArrayList<ApplicationBasicInfo> appInfo = new ArrayList<>();
-        for (ApplicationBasicInfo applicationBasicInfo: applicationBasicInfos) {
-            if (ApplicationMgtUtil.isUserAuthorized(applicationBasicInfo.getApplicationName(), getUsername())) {
-                appInfo.add(applicationBasicInfo);
-                if (log.isDebugEnabled()) {
-                    log.debug("Application Name:" + applicationBasicInfo.getApplicationName());
+    public ApplicationBasicInfo[] getAllApplicationBasicInfo() throws IdentityApplicationManagementException {
+
+        try {
+            applicationMgtService = ApplicationManagementService.getInstance();
+            ApplicationBasicInfo[] applicationBasicInfos =
+                    applicationMgtService.getAllApplicationBasicInfo(getTenantDomain(), getUsername());
+            ArrayList<ApplicationBasicInfo> appInfo = new ArrayList<>();
+            for (ApplicationBasicInfo applicationBasicInfo : applicationBasicInfos) {
+                if (ApplicationMgtUtil.isUserAuthorized(applicationBasicInfo.getApplicationName(), getUsername())) {
+                    appInfo.add(applicationBasicInfo);
+                    if (log.isDebugEnabled()) {
+                        log.debug("Application Name: " + applicationBasicInfo.getApplicationName());
```

Comment:
Better to improve the log as something like "Retrieving basic information of application: "